smart-home-application/include/UniquePointer.h

```diff
@@ -0,0 +1,33 @@
+#ifndef SHA_UNIQUEPOINTER_H_
+#define SHA_UNIQUEPOINTER_H_
+
+#include <memory>
+
+namespace up {
+
+    template<typename T>
+    class UniquePointer {
```

Comment:
unique_ptr class has the ability to use a custom deleter as well, please double check the definition of [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr)